site/app/entities/course/CourseMaterial.php

```diff
@@ -0,0 +1,175 @@
+<?php
+
+namespace app\entities\course;
+
+use app\libraries\DateUtils;
+use app\models\User;
+use Doctrine\Common\Collections\ArrayCollection;
+use Doctrine\Common\Collections\Collection;
+use Doctrine\ORM\Mapping as ORM;
+
+/**
+ * Class CourseMaterial
+ * @package app\entities
+ * @ORM\Entity
+ * @ORM\Table(name="course_materials")
+ */
+class CourseMaterial {
+    const FILE = 0;
+    const LINK = 1;
+
+    /**
+     * @ORM\Id
+     * @ORM\Column(type="integer")
+     * @ORM\GeneratedValue(strategy="AUTO")
+     * @var integer
+     */
+    protected $id;
+
+    /**
+     * @ORM\Column(type="string")
+     * @var string
+     */
+    protected $path;
+
+    /**
+     * @ORM\Column(type="smallint")
+     * @var int
+     */
+    protected $type;
+
+    /**
+     * @ORM\Column(type="datetimetz")
+     * @var \DateTime
+     */
+    protected $release_date;
+
+    /**
+     * @ORM\Column(type="boolean")
+     * @var boolean
+     */
+    protected $hidden_from_students;
+
+    /**
+     * @ORM\Column(type="float")
+     * @var float
+     */
+    protected $priority;
+
+    /**
+     * @ORM\OneToMany(targetEntity="\app\entities\course\CourseMaterialSection", mappedBy="course_material", fetch="EAGER", cascade={"persist"}, orphanRemoval=true)
+     * @var Collection<CourseMaterialSection>
+     */
+    protected $sections;
+
+    public function __construct(array $details) {
+        $this->setType($details['type']);
+        $this->setPath($details['path']);
+        $this->setReleaseDate($details['release_date']);
+        $this->setHiddenFromStudents($details['hidden_from_students']);
+        $this->setPriority($details['priority']);
+        $this->sections = new ArrayCollection();
+    }
+
+    /**
+     * @return Collection
+     */
```

Comment:
`Collection` in Doctrine is a "generic" type, which allows for specifying what it has by using the `<>` notation within the docstring. In this case, it would be `Collection<CourseMaterialSection>`, and this then allows phpstan to be aware of this info on static analysis.